x-pack/plugins/security_solution/public/management/pages/blocklist/view/components/create_blocklist_form.tsx

```diff
@@ -0,0 +1,464 @@
+            },
+          ],
+        },
+      });
+    },
+    [validate, blocklistEntry, onChange, item]
+  );
+
+  const handleOnFieldChange = useCallback(
+    (field: ConditionEntryField) => {
+      const nextIsValid = validate(field, selectedOs, blocklistEntry.type, blocklistEntry.value);
+
+      onChange({
+        isValid: nextIsValid,
+        item: {
+          ...item,
+          entries: [{ ...blocklistEntry, field }],
+        },
+      });
+    },
+    [validate, onChange, item, selectedOs, blocklistEntry]
+  );
+
+  // only triggered on remove / clear
+  const handleOnValueChange = useCallback(
+    (change: Array<EuiComboBoxOptionOption<string>>) => {
+      const value = change.map((option) => option.label);
+
+      const nextIsValid = validate(blocklistEntry.field, selectedOs, blocklistEntry.type, value);
+
+      onChange({
+        isValid: nextIsValid,
+        item: {
+          ...item,
+          entries: [{ ...blocklistEntry, value }],
+        },
+      });
+    },
+    [validate, onChange, item, selectedOs, blocklistEntry]
+  );
+
+  const handleOnValueAdd = useCallback(
+    (option: string) => {
+      const splitValues = option.split(',').filter((value) => value.trim());
+      const value = [...blocklistEntry.value, ...splitValues];
+
+      const nextIsValid = validate(blocklistEntry.field, selectedOs, blocklistEntry.type, value);
+
+      onChange({
+        isValid: nextIsValid,
+        item: {
+          ...item,
+          entries: [{ ...blocklistEntry, value }],
+        },
+      });
+    },
+    [validate, onChange, item, selectedOs, blocklistEntry]
+  );
+
+  const handleOnPolicyChange = useCallback(
+    (change: EffectedPolicySelection) => {
+      const tags = change.isGlobal
+        ? [GLOBAL_ARTIFACT_TAG]
+        : change.selected.map((policy) => `${BY_POLICY_ARTIFACT_TAG_PREFIX}${policy.id}`);
+
+      onChange({
+        isValid,
+        item: {
+          ...item,
+          tags,
+        },
+      });
+    },
+    [onChange, isValid, item]
+  );
+
+  return (
+    <EuiForm component="div">
+      <EuiText size="xs">
+        <h3>{DETAILS_HEADER}</h3>
+      </EuiText>
+      <EuiSpacer size="xs" />
+      <EuiText size="s">
+        <p>{DETAILS_HEADER_DESCRIPTION}</p>
+      </EuiText>
+      <EuiSpacer size="m" />
+
+      <EuiFormRow
+        label={NAME_LABEL}
+        isInvalid={!!errors.name?.length}
+        error={errors.name}
+        fullWidth
+      >
+        <EuiFieldText
+          name="name"
+          value={item.name}
+          onChange={handleOnNameChange}
+          onBlur={handleOnNameBlur}
+          required={visited.name}
+          maxLength={256}
+          fullWidth
+        />
+      </EuiFormRow>
+      <EuiFormRow label={DESCRIPTION_LABEL} fullWidth>
+        <EuiTextArea
+          name="description"
+          value={item.description}
+          onChange={handleOnDescriptionChange}
+          fullWidth
+          compressed
+          maxLength={256}
```

Comment:
we should ensure the API also has these limitation built into it, since the UI seems to be enforcing them. Can you look at the API validation issue and add if not there - or mabye we don't restrict? (what do we do for Event Filters and Exceptions?)

Reply:
good point. we're sure we want these limits right? I just copied these over from trusted apps but they make sense to me.

Reply:
Trusted Apps is not a good one to copy patterns from. Maybe use Event Filters. But, I just looked at the API validations and we seem to already do [this validation](https://github.com/elastic/kibana/blob/9d53810791165e9a16fa7dc6aac8ca4179b26b52/x-pack/plugins/security_solution/server/lists_integration/endpoint/validators/base_validator.ts#L25-L28) there (sorry, I forgot), so I think you are good here. Sorry for the noise.